docs/programming-oak.md

```diff
@@ -368,20 +368,46 @@ using the following script:
 ./scripts/push_example -e hello_world
 ```
 
+### Creating a Permissions File
+
+A permissions file is a `.toml` file provided by the host owner, as an
+additional layer of defense. This file specifies the features of Oak that are
+permitted for the applications running on the host. In particular, using this
+file, the host owner can enable or disable gRPC or HTTP connections. In
+addition, it is possible to specify an allowlist of external gRPC or HTTP
+services that the applications can connect to. Interaction with all other
+external servers will be prohibited. Connections to insecure HTTP servers will
+be allowed only if explicitly enabled via the `allow_insecure_http_egress` flag.
```

Comment:
Does this mean that even if an http URL is present in `allow_egress_addresses`, it is not actually honored, if this flag is off? It seems unnecessary, since those URLs already have a schema?

Reply:
This flag concerns insecure HTTP clients, where the authority is empty. We decided earlier that it does not make sense to provide an authority in case of an HTTP client for insecure connections. The URIs/authorities in `allow_egress_addresses` will only be used for gRPC and secure HTTP connections. So, the list will still be honored, even if `allow_egress_addresses` is off.